src/rtl/inc/xdpassert.h

```diff
@@ -23,8 +23,15 @@
 #define ASSERT(e) NT_ASSERT_ASSUME(e)
 #endif
 #else
+#if __SANITIZE_ADDRESS__
+// Instead of asserting (which simply exits the process) generate an AV that ASAN will catch.
+#define ASSERT_ACTION ((*(UINT16 *)0xDEAD) = 0xDEAD)
```

Comment:
Why is this necessary? We don't do this for MsQuic.

Reply:
The comment directly above tried to explain the problem this solves: regular assertion handling kills the process without the benefit of an ASAN callstack, an ASAN crash dump, etc. Maybe MsQuic solved this a different way, or doesn't have this problem.

Reply:
What is an ASAN crash dump? A dump is a dump? We just grab any crash dump triggered.

Reply:
ASAN produces crash dumps when the `ASAN_SAVE_DUMPS` environment variable is set, and it prints out a callstack and the input that caused the crash. I'm not sure where the other crash dumps end up - maybe that depends on WER? I don't want to mess with those settings on dev boxes.

Reply:
Ah, OK, I see QUIC creates WER registry keys for each process that it creates. That makes sense but requires admin privileges - I would like our user mode tests to not require admin.